lib/wiki_api.rb

```diff
@@ -51,51 +51,56 @@ def get_article_rating(titles)
     titles = [titles] unless titles.is_a?(Array)
     titles = titles.sort_by(&:downcase)
 
-    talk_titles = titles.map { |title| 'Talk:' + title }
-    raw = get_raw_page_content(talk_titles)
-    return [] unless raw
-
-    # Pages that are missing get returned before pages that exist, so we cannot
-    # count on our array being in the same order as titles.
-    raw.map do |_article_id, talkpage|
-      # Remove "Talk:" from the "title" value to get the title.
-      { talkpage['title'][5..-1].tr(' ', '_') =>
-        parse_article_rating(talkpage) }
-    end
-  end
-
-  ###################
-  # Parsing methods #
-  ###################
-
-  def parse_article_rating(raw_talk)
-    # Handle MediaWiki API errors
-    return nil if raw_talk.nil?
-    # Handle the case of nonexistent talk pages.
-    return nil if raw_talk['missing']
-
-    wikitext = raw_talk['revisions'][0]['*']
-    ArticleClassExtractor.new(wikitext).extract
+    query_parameters = { titles: titles,
+                         prop: 'pageassessments', 
+                         formatversion: '2',
+                         redirects: 'true' }
+    response = get_responses_continue(query_parameters)
+    return [] unless response
+    return [] unless response['query'] 
+
+    info = response['query']['pages']
```

Comment:
Instead of these guard clauses, we can use `.dig` to safely retrieve the expected `pages` value.